### ISSUE TITLE
0MH/s, even when benchmarking

### ISSUE THREAD

**Issue by nicolaevladescu** [ethereum-mining/ethminer#95]
Hello.

I currently built a successful binary on `ubuntu 16.04.2 amd64` fresh install.

I had a library error during build that i fixed installing  `mesa-common-dev`

The issue is that i always have 0MH/s and it never generated a DAG, hangs at `got work package` and never gets more than 0MH/s

Do i still new the AMD APP SDK when building ?

If so, how do i specify the path to it via build options?

My GPU is a Gigabyte RX 580

Thank you.

```
[OPENCL]:Found suitable OpenCL device [Ellesmere] with 3221225472 bytes of GPU memory
  m  21:12:51|ethminer  Getting work package...
  m  21:12:51|ethminer  Got work package: #36b45916
  ℹ  21:12:51|openclminer0  set work; seed: #9e972470, target:  #0000000225c1
  ℹ  21:12:51|openclminer0  Initialising miner...
  m  21:12:52|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:52|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:53|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:53|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:53|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
[OPENCL]:Using platform: AMD Accelerated Parallel Processing
[OPENCL]:Using device: Ellesmere(OpenCL 1.2 AMD-APP (2348.3))
[OPENCL]:clCreateCommandQueue(-6)
[OPENCL]:clEnqueueWriteBuffer(-36)
  m  21:12:54|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:54|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:55|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:55|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:56|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:56|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:57|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:57|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:57|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:58|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:58|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:59|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:12:59|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:00|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:00|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:00|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:01|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:01|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:02|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:02|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:03|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:03|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:03|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:04|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:04|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:05|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:05|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:06|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:06|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:06|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:07|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:07|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
m  21:13:08|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:08|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:09|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:09|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:09|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:10|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:10|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:11|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:11|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:12|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:12|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:13|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:13|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:13|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:14|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:14|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:15|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:15|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:16|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:16|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:16|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:17|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:17|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:18|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:18|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:19|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:19|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:19|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:20|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:20|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:21|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:21|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:22|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:22|ethminer  Mining on PoWhash #36b45916 : 0.00MH/s [A0+0:R0+0:F0]
  m  21:13:22|ethminer  Got work package: #76b0ccac
```

**Comment by Genoil** [comment]
the error (-6) means CL_OUT_OF_HOST_MEMORY.  you should set these variables before starting ethminer:

```
export GPU_FORCE_64BIT_PTR=0 
export GPU_MAX_HEAP_SIZE=100 
export GPU_USE_SYNC_OBJECTS=1 
export GPU_MAX_ALLOC_PERCENT=100 
export GPU_SINGLE_ALLOC_PERCENT=100
```

**Comment by nicolaevladescu** [comment]
Hello, i am using exactly this vars and values and an extra `export DISPLAY=:0`, could it be a host RAM issue ? I only have 8G of RAM on that MB.

I installed the official `cpp-ethereum` and it starts to generate the DAG and finished then gives the same two errors.

```
Creating DAG. 99% done...DAG  21:27:54.007|ethminer  Generating DAG file. Progress: 100 %
Creating DAG. 100% done...miner  21:27:54.012|ethminer  Got work package:
miner  21:27:54.012|ethminer    Header-hash: d62d13d7c0553900e3b297ece53f6bceb67017c785e41e20fd7a75e16ff97c1e
  ℹ  21:27:54.012|ethminer  Loading full DAG of seedhash: #68fc811a…
miner  21:27:54.012|ethminer    Seedhash: 9e9724706b96cf93985fa8f097f4c1a687ebc3c5fc93d50a68d8e51211a723e0
miner  21:27:54.012|ethminer    Target: 0000000225c17d04dad2965cc5a02a23e254c0c3f75d9178046aeb27ce1ca574
  ℹ  21:27:54.012|gpuminer0  workLoop 0 #00000000… #9e972470…
  ℹ  21:27:54.013|gpuminer0  Initialising miner...
miner  21:27:54.213|ethminer  Mining on PoWhash #d62d13d7… : 0 H/s = 0 hashes / 0.2 s
[OPENCL]:Using platform: AMD Accelerated Parallel Processing
[OPENCL]:Using device: Ellesmere(OpenCL 1.2 AMD-APP (2348.3))
miner  21:28:08.752|ethminer  Got work package:
miner  21:28:08.753|ethminer    Header-hash: 80765761417021043085d2eb0c53bd2ee995d207bd91a5cfa43e20ec80c3a172
miner  21:28:08.753|ethminer    Seedhash: 9e9724706b96cf93985fa8f097f4c1a687ebc3c5fc93d50a68d8e51211a723e0
miner  21:28:08.753|ethminer    Target: 0000000225c17d04dad2965cc5a02a23e254c0c3f75d9178046aeb27ce1ca574
DAG  21:28:08.753|ethminer  Generating DAG file. Progress: 0 %
[OPENCL]:clCreateCommandQueue(-6)
[OPENCL]:clEnqueueWriteBuffer(-36)
DAG  21:28:25.146|ethminer  Generating DAG file. Progress: 1 %
```

I am also running from and USB stick could it be too slow for generating a DAG then it's outdated and has to do it again?

**Comment by nicolaevladescu** [comment]
I switched driver from `amdgpu-pro 17.10` to `amdgpu 16.10` and nothing changed, also tried with a r9 380 4GB, and RX 550 4GB and nothing changed again.

I was thinking i may have missed something going from server with minimal GUI, so i tried with Lubuntu 16.04 and the same issue.

**Comment by derubm** [comment]
Did i understand it wrong that the dag file is created in RAM entirely ? Anyway, try the pre compiled versions and compare results.

**Comment by nicolaevladescu** [comment]
Yeah i thought that too, but was worth asking. The precompiled versions didn't work, they go further in generating a DAG but that's it.

It might be an `amdgpu-pro 17.10` bug, or hardware one.

Found this in `dmesg`

```
user@miner01:~$ dmesg | grep powerplay
[    2.656121] amdgpu: [powerplay] amdgpu: powerplay sw initialized
[    2.709710] amdgpu: [powerplay] Can't find requested voltage id in vdd_dep_on_sclk table!
```

**Comment by nicolaevladescu** [comment]
Unfortunately i could not get to the bottom of this, i moved to windows and it works. I suspect a driver/kernel module issue since the same hardware works with windows.

Thank you everybody that replied.

**Comment by nicolaevladescu** [comment]
Went back and fixed it with `amdgpu-pro-16.40`

`[    2.709710] amdgpu: [powerplay] Can't find requested voltage id in vdd_dep_on_sclk table!`

Was gone.

NOTE: for some `xserver-xorg` reasons i have yet to discover, it works only with `sudo`
